### PR TITLE
moleculer microservice /nodejs

### PR DESCRIPTION
**Pull request by bashiru98**
added nodejs assignment please check for review

### CODE REVIEW

File: moleculer-microservice/package.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    "jest": "^25.1.0",
+    "jest-cli": "^25.1.0",
+    "moleculer-repl": "^0.6.2",
+    "supertest": "^6.1.3",
+    "ts-jest": "^25.3.0",
+    "ts-node": "^8.8.1"
+  },
+  "dependencies": {
+    "@elastic/elasticsearch": "^7.13.0",
+    "@types/jest": "^25.1.4",
```
Comment (dhavalsoni9989) [inline moleculer-microservice/package.json:44]: @types package should be in devDependencies

File: moleculer-microservice/package.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    "@types/mkdirp": "^1.0.0",
```
Comment (dhavalsoni9989) [inline moleculer-microservice/package.json:45]: @types package should be in devDependencies

File: moleculer-microservice/package.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    "@types/node": "^13.9.8",
```
Comment (dhavalsoni9989) [inline moleculer-microservice/package.json:46]: @types package should be in devDependencies

File: moleculer-microservice/package.json [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    "install": "^0.13.0",
+    "ioredis": "^4.27.6",
+    "jsonwebtoken": "^8.5.1",
+    "moleculer": "^0.14.0",
+    "moleculer-db": "^0.8.14",
+    "moleculer-db-adapter-mongo": "^0.4.12",
+    "moleculer-elasticsearch": "^1.1.6",
+    "moleculer-web": "^0.9.0",
+    "nats": "^1.3.2",
+    "npm": "^7.20.3",
```
Comment (dhavalsoni9989) [inline moleculer-microservice/package.json:60]: why add npm package ?

File: moleculer-microservice/package.json [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    "redis": "^3.1.2",
+    "redlock": "^4.2.0",
+    "ts-mixer": "^5.2.1",
+    "typeorm": "^0.2.36",
```
Comment (dhavalsoni9989) [inline moleculer-microservice/package.json:64]: I did not see any use of this package in this project

File: moleculer-microservice/package.json [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    "typescript": "^3.8.3",
```
Comment (dhavalsoni9989) [inline moleculer-microservice/package.json:65]: typescript package should be in devDependencies

File: moleculer-microservice/package.json (hunk 5 above)
Comment (bashiru98) [reply moleculer-microservice/package.json:64]: ooh sorry i used it in another project that was why let me fix it

File: moleculer-microservice/package.json (hunk 6 above)
Comment (bashiru98) [reply moleculer-microservice/package.json:65]: alright yh i got it

File: moleculer-microservice/package.json (hunk 3 above)
Comment (bashiru98) [reply moleculer-microservice/package.json:46]: yh sure

File: moleculer-microservice/package.json (hunk 1 above)
Comment (bashiru98) [reply moleculer-microservice/package.json:44]: sure

**Comment by bashiru98** [comment]
i have fixed the issues in the above comments

File: moleculer-microservice/services/authentication.service.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+							{ ...userData }
+						);
+						return await new Register(entity, { ...user }).$handler(
+							ctx,
+							user
+						);
+					},
+				},
+				login: {
+					rest: { ...Login.rest },
```
Comment (dhavalsoni9989) [inline moleculer-microservice/services/authentication.service.ts:52]: You can use the default authorize method of moleculer for login purpose. ref: https://moleculer.services/docs/0.14/moleculer-web.html#Authorization

File: moleculer-microservice/src/api/login.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+import { Mixin } from "ts-mixer";
+
+import { Errors } from "moleculer"
+
+import { Token } from "../utils/token";
+import { UserPayload } from "../utils/types";
+import { UserAction } from "../services/users/index";
+import { Password } from "../utils/password";
+
+export class Login extends Mixin(Password, Token, UserAction) {
```
Comment (dhavalsoni9989) [inline moleculer-microservice/src/api/login.ts:14]: It's not required need to use moleculer default authorize methods

ref : https://moleculer.services/docs/0.14/moleculer-web.html#Authorization

File: moleculer-microservice/mixins/db.mixin.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+"use strict";
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-expect-error
+import MongoAdapter from "moleculer-db-adapter-mongo";
+import { Service, ServiceSchema } from "moleculer";
+import DbService from "moleculer-db";
+
+export default class Connection
```
Comment (dhavalsoni9989) [inline moleculer-microservice/mixins/db.mixin.ts:11]: User and all information are stored in elastic search then why used moleculer-db and defined the db mixins

Comment (bashiru98) [reply moleculer-microservice/mixins/db.mixin.ts:11]: i used the database to fetch dummy products for testing purpose

File: moleculer-microservice/services/authentication.service.ts (hunk 7 above)
Comment (bashiru98) [reply moleculer-microservice/services/authentication.service.ts:52]: alright that sounds cool

File: moleculer-microservice/src/api/login.ts (hunk 8 above)
Comment (bashiru98) [reply moleculer-microservice/src/api/login.ts:14]: alright cool

**Pull request closed without merging**